src/chat/summarizers/connections.ts

```diff
@@ -0,0 +1,133 @@
+import { MarkdownString } from "vscode";
+import {
+  CCloudStatus,
+  Connection,
+  ConnectionType,
+  KafkaClusterConfig,
+  KafkaClusterStatus,
+  LocalConfig,
+  SchemaRegistryConfig,
+  SchemaRegistryStatus,
+} from "../../clients/sidecar";
+import { ContextValues, getContextValue } from "../../context/values";
+
+/** Create a string representation of a {@link Connection} object. */
+export function summarizeConnection(connection: Connection): string {
+  const type: ConnectionType = connection.spec.type!;
+  let summary = new MarkdownString().appendMarkdown(`- "${connection.spec.name}"`);
+
+  // add spec/status details depending on the connection type
+  switch (type) {
+    case ConnectionType.Ccloud: {
+      summary = summarizeCCloudConnection(connection, summary);
+      break;
+    }
+    case ConnectionType.Local: {
+      summary = summarizeLocalConnection(connection, summary);
+      break;
+    }
+    case ConnectionType.Direct: {
+      summary = summarizeDirectConnection(connection, summary);
+      break;
+    }
+  }
+  return summary.value;
+}
+
+/**
+ * Summarize the `CCLOUD` {@link Connection}.
+ * @param connection The {@link Connection} object to summarize.
+ * @param summary The {@link MarkdownString} to append to.
+ * @returns The updated {@link MarkdownString}.
+ */
+export function summarizeCCloudConnection(
```

Comment:
should we include org/env in the summary?

Reply:
The org could be handy, but I'd like to leave anything at the environment level (or below, into Kafka/SR/etc) for the next tool

Reply:
Update: https://github.com/confluentinc/vscode/pull/1755 prepped